frame/examples/offchain-worker-ping-pong/src/lib.rs

```diff
@@ -0,0 +1,632 @@
+				pong_payload: ref payload,
+				ref signature,
+			} = call
+			{
+				// ⚠ WARNING ⚠
+				// this is nothing but a "sanity check" on the signature
+				// it only checks if the signature is coherent with the public key of `SignedPayload`
+				// whoever that might be (not necessarily an authorized OCW)
+				let signature_valid =
+					SignedPayload::<T>::verify::<T::AuthorityId>(payload, signature.clone());
+				if !signature_valid {
+					return InvalidTransaction::BadProof.into();
+				}
+				Self::validate_transaction_parameters(&payload.block_number)
+			} else if let Call::pong_unsigned { block_number, nonce: _n } = call {
+				Self::validate_transaction_parameters(block_number)
+			} else {
+				InvalidTransaction::Call.into()
+			}
+		}
+	}
+
+	/// A struct for wrapping the ping nonce.
+	#[derive(Encode, Decode, MaxEncodedLen, TypeInfo)]
+	pub struct Ping(pub u32);
+
+	/// A vector of recently submitted pings.
+	#[pallet::storage]
+	#[pallet::getter(fn pings)]
+	pub(super) type Pings<T: Config> = StorageValue<_, BoundedVec<Ping, T::MaxPings>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn authorities)]
+	pub(super) type Authorities<T: Config> =
+		StorageValue<_, BoundedVec<T::AccountId, T::MaxAuthorities>, ValueQuery>;
+
+	/// Defines the block when next unsigned transaction will be accepted.
+	///
+	/// To prevent spam of unsigned (and unpaid!) transactions on the network,
+	/// we only allow one transaction every `T::UnsignedInterval` blocks.
+	/// This storage entry defines when new transaction is going to be accepted.
+	#[pallet::storage]
+	#[pallet::getter(fn next_unsigned_at)]
+	pub(super) type NextUnsignedAt<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+}
+
+/// Payload used by this example crate to hold pong response
+/// data required to submit an unsigned transaction.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, scale_info::TypeInfo)]
+pub struct PongPayload<Public, BlockNumber> {
+	block_number: BlockNumber,
+	nonce: u32,
+	public: Public,
+}
+
+impl<T: SigningTypes> SignedPayload<T> for PongPayload<T::Public, T::BlockNumber> {
+	fn public(&self) -> T::Public {
+		self.public.clone()
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	fn is_authority(who: &T::AccountId) -> bool {
+		<Authorities<T>>::get().contains(who)
+	}
+
+	fn validate_transaction_parameters(block_number: &T::BlockNumber) -> TransactionValidity {
+		// Now let's check if the transaction has any chance to succeed.
+		let next_unsigned_at = <NextUnsignedAt<T>>::get();
+		if &next_unsigned_at > block_number {
+			return InvalidTransaction::Stale.into();
+		}
+		// Let's make sure to reject transactions from the future.
+		let current_block = <system::Pallet<T>>::block_number();
+		if &current_block < block_number {
+			return InvalidTransaction::Future.into();
+		}
+
+		ValidTransaction::with_tag_prefix("ExampleOffchainWorker")
+			// We set base priority to 2**20 and hope it's included before any other
```

Comment:
Where is this 2**28 happening?

Reply:
https://github.com/paritytech/substrate/blob/77e792969cfca5059796a5b64bb7439c32daead9/frame/examples/offchain-worker-ping-pong/src/tests.rs#L112